src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1967,6 +1972,16 @@ ch switch
                 _ => $"\\u{(uint)ch:X4}"
             };
 
+        public static string DescribeString(string s)
```

Comment:
The only call site to this is one that appends the whole result to a StringBuilder... rather than create a new StringBuilder, convert that to a string, and then append the resulting string to another builder, just make this method void and pass in the builder to append to directly.